### ISSUE TITLE
Since we can get duplicate/repetitive booking records, ConcurrencyFailureException add this as well 

### ISSUE THREAD

**Issue by amit5881** [nitsshukla/AgodaBooking#4]
https://github.com/nitsshukla/AgodaBooking/blob/24095baf0c394ba4ae17d8e56d237e8eaaa461ff/src/main/java/com/agoda/booking/tracker/service/BookingService.java#L25

**Comment by nitsshukla-zz** [comment]
Good catch, but I think I don't need to retry for repetitive booking (same booking-id)

I'm thinking, how can I tackle this situation:
1) saved a booking with booking-id: 1234
2) Got a request for 3 bookings with I'd: 1234,1235,1236

Now during saveAll I need to save only 1235,1236
